File: Projects/pa3/DictionaryTest.java
/*	Nicholas Pappas
	nhpappas@ucsc.edu
	ID: 1558554
	Assignment: pa3
*/
//-----------------------------------------------------------------------------
// DictionaryTest.java
// Test Client for the Dictionary class
//-----------------------------------------------------------------------------

public class DictionaryTest{

   public static void main(String[] args){
	  String print;
      Dictionary A = new Dictionary();

      A.insert("1","does"); //Start with making sure the Dictionary is loadable
      A.insert("2","this"); //Check intial edge cases first
      A.insert("3","work"); //Had to fix the findKey() to get it to work properly
      A.insert("4","yes");
      A.insert("5","it");
      A.insert("6","does");
      A.insert("7","g");
      System.out.println(A);
	  System.out.println("size= " + A.size()); //Check size()

	  print = A.lookup("3"); //Check lookup() returns properly for value != null
      System.out.println("key=3 "+(print==null?"not found":("value="+print)));
	  System.out.println("isEmpty() is called: " + A.isEmpty()); //check boolean case false
	  
	  System.out.println("makeEmpty() is called"); //Test methods in empty state
	  A.makeEmpty();
	  System.out.println("isEmpty() is called: " + A.isEmpty()); //check bollean case true
	  System.out.println("size= " + A.size());
	  print = A.lookup("3");
      System.out.println("key=3 "+(print==null?"not found":("value="+print)));
	  
	  A.insert("breakfast", "first"); //After making empty add items, test again
	  A.insert("dinner", "2nd");
	  A.insert("lunch", "third");
	  System.out.println(A);
	  System.out.println("size= " + A.size());
	  //System.out.println("Try to insert lunch twice");
	  //A.insert("lunch", "third"); //Test DuplicateKeyException
	  
	  System.out.println("delete lunch and dinner");
	  A.delete("dinner"); //Test delete
	  A.delete("lunch");
	  System.out.println(A);
	  System.out.println("size= " + A.size());
	  //System.out.println("Try to delete dinner again");
	  //A.delete("dinner"); //Test for KeyNotFoundException
	  
	  
	  
   }
}

File: Labs/lab5/DictionaryTest.c
/*	Nicholas Pappas
	nhpappas@ucsc.edu
	ID: 1558554
	Assignment: pa3
*/
//-----------------------------------------------------------------------------
// DictionaryTest.c
// Test Client for the Dictionary ADT
//-----------------------------------------------------------------------------

#include<stdio.h>
#include<stdlib.h>
#include<string.h>
#include<assert.h>
#include"Dictionary.h"

#define MAX_LEN 180

int main(int argc, char* argv[]){
   Dictionary A = newDictionary();
   char* k;
   char* v;
   insert(A, "1","does"); //Start with making sure the Dictionary is loadable
 //delete(A, "1"); //Check the edge case delete head when it is the only node
   
 //insert(A, "1", "does"); // error: key collision
   insert(A, "2","this"); //I had to debug lookup() and findkey() so that insert() complied
   insert(A, "3","work");
   insert(A, "4","yes"); 
   insert(A, "5","it");
   insert(A, "6","does");
   insert(A, "7","!");
   
 /*printDictionary(stdout, A);*/ //cut&paste the left "/*" to easily comment out all but one pair
 //printf("size= %d\n", size(A)); //Check size() 
   printf("This Dictionary contains: %d elements\n", size(A));
   printf("Dictionary is empty: %s\n", (isEmpty(A)?"true":"false"));
   
   printf("Call lookup(Dictionary, key).\n");
   k = "1";
   v = lookup(A, k);
   printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   k = "2";
   v = lookup(A, k);
   printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   k = "3";
   v = lookup(A, k);
   printf("key=\"%s\" %s\"%s\"\n", k, (v==NULL?"not found ":"value="), v);
   
   printf("Delete first three elements.\n"); //It's essential to check edge cases. Memory was leaking from delete.
   delete(A, "1"); //Head
   delete(A, "3"); //Middle
   delete(A, "7"); //Tail
 //delete(A, "1");  // error: key not found
   printf("This Dictionary contains: %d elements\n", size(A));
   printDictionary(stdout, A);

   printf("Call makeEmpty(Dictionary).\n");
   makeEmpty(A);
   printf("Dictionary is empty: %s\n", (isEmpty(A)?"true":"false"));

   freeDictionary(&A); //1st thing: had to fix makeEmpty() to get rid of segmentation fault

   return(EXIT_SUCCESS);
}